temporal/server_option.go

```diff
@@ -188,3 +191,24 @@ func WithCustomMetricsHandler(provider metrics.MetricsHandler) ServerOption {
 		s.metricHandler = provider
 	})
 }
+
+type fxOptions []fx.Option
+
+func (o fxOptions) apply(s *serverOptions) {}
+func (o fxOptions) fxOptions() []fx.Option { return o }
+
+// WithFxOptions passes a list of options to the underlying fx application. You can use this to do things like override
+// any dependency that the server needs. For example:
+//
+//	server.WithFxOptions(fx.Supply(myCustomConfig))
+//
+// This will override the default config with your custom config.
+// Additionally, you can decorate existing dependencies. For example:
+//
+//	server.WithFxOptions(fx.Decorate(func(cfg *config.Config) *config.Config {
+//		cfg.Persistence.NumHistoryShards = 10
+//		return cfg
+//	}))
+func WithFxOptions(opts ...fx.Option) ServerOption {
+	return fxOptions(opts)
+}
```

Comment:
can you follow the same pattern as other server options (return applyFunc)

Reply:
The `applyFunc` is execued while fx is constructing objects, after it constructs the graph. However, we need these options to modify the graph itself, so we have to provide them to fx before that.